flask/db.py

```diff
@@ -36,49 +35,70 @@
 def get_products():
     results = []
     try:
-        connection = db.connect()
-        products = connection.execute(
-            "SELECT * FROM products"
-        ).fetchall()
-        
-        for product in products:
-            reviews = connection.execute(
-                "SELECT * FROM reviews WHERE productId = {}".format(product.id)
+        with sentry_sdk.start_span(op="get_products", description="db.connect"):
+            connection = db.connect()
+
+        with sentry_sdk.start_span(op="get_products", description="db.query") as span:
+            products = connection.execute(
+                "SELECT * FROM products"
             ).fetchall()
-            result = dict(product)
-            result["reviews"] = []
+            span.set_tag("totalProducts",len(products))
+            span.set_data("products",products)
+        
+        with sentry_sdk.start_span(op="get_products.reviews", description="db.query") as span:
+            for product in products:
+                reviews = connection.execute(
+                    "SELECT * FROM reviews WHERE productId = {}".format(product.id)
+                ).fetchall()
+                result = dict(product)
+                result["reviews"] = []
+
+                for review in reviews:
+                    result["reviews"].append(dict(review))
+                results.append(result)
+            span.set_data("reviews", results)
 
-            for review in reviews:
-                result["reviews"].append(dict(review))
-            results.append(result)
-        return json.dumps(results, default=str)
+        with sentry_sdk.start_span(op="serialization", description="json"):
```

Comment:
Serialization is known to cause performance problems (delays) so capture as a span.

it's explained here:
https://www.notion.so/sentry/Performance-Use-Cases-0a788c41f1104545bd41921d2e3bb416#be9898339ce94cc19765306e46758153